sys/include/net/nanocoap.h

```diff
@@ -1500,6 +1500,61 @@ static inline ssize_t coap_opt_add_uri_path_buffer(coap_pkt_t *pkt,
     return coap_opt_add_chars(pkt, COAP_OPT_URI_PATH, path, path_len, '/');
 }
 
+/**
+ * @brief   Adds an entity-tag (etag) value to the coap pkt.
+ *
+ * @param[in,out] pkt         Packet being built
+ * @param[in]     etag        Etag value of the packet
+ * @param[in]     len         Length of @p etag in bytes
+ *
+ * @return        number of bytes written to pkt buffer
+ * @return        -EINVAL if invalid etag size is used for @p len
+ * @return        -ENOSPC if no available options or pkt full
+ */
+static inline ssize_t coap_opt_add_etag(coap_pkt_t *pkt, const uint8_t *etag, size_t len)
```

Comment:
```suggestion
static inline ssize_t coap_opt_add_etag(coap_pkt_t *pkt, const void *etag, size_t len)
```

If `etag` is opaque, this is more ergonomic

Reply:
True, fixed!